Sources/App/Interactors/ParsePerformanceTestResultsInteractor.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+public struct ParsePerformanceTestResultsInteractor {
+    enum OutputParsingError: Error {
+        case missingTestCases
+    }
+    
+    public func execute(output: String) throws -> [PerformanceTestResults] {
+        let replaced = output.replacingOccurrences(of: "[PERFORMANCE]", with: "🔤")
```

Comment:
Maybe something like this could be better? Not tested, but I think the idea would work, and it doesn't need any regex.
```swift
let averageTimes: [String] = output.split(separator: "\n")
    .filter { $0.contains("measured [Time") }
    .map { $0.split(separator: " ")[7] }
    .map(String.init)

Reply:
Updated to something similar.

```swift
let doubleChars = [(UInt32("0")...UInt32("9")), (UInt32(".")...UInt32("."))].joined().map { Character(UnicodeScalar($0)!) }

let split = output.split(separator: "\r\n")
let filterNonExpected = split.filter { $0.contains("[PERFORMANCE] test") }
let filterNonPerformance = split.filter { $0.contains("measured [Time") }
let expectedResults = filterNonExpected
    .map { $0.split(separator: " ")[3] }
    .map { str in str.filter { doubleChars.contains($0) } }
    .map { Double($0)! }
let results = filterNonPerformance
    .map { $0.split(separator: " ")[8] }
    .map { str in str.filter { doubleChars.contains($0) } }
    .map { Double($0)! }
let names = filterNonExpected
    .map { $0.split(separator: " ")[1] }
    .map { funcName in funcName.filter { ("A"..."z").contains($0) } }
    .map { String($0) }

let testResults = zip(names, zip(results, expectedResults))
    .map { name, stats -> (name: String, expected: Double, average: Double, change: String) in
        let (average, expected) = stats
        
        return (
            name: name,
            expected: expected,
            average: average,
            change: "\(String(format:"%.2f", Double((expected - average))/expected * 100))%"
        )
    }
```